### ISSUE TITLE
adding/changing comments fail in SOLR call

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud#2234]
Follow-up to #2225 - once the visibility check was fixed, I discovered a couple of other bugs.

@prateek05, it looks like the format for the structure we're saving comments in SOLR has changed. Both adding a new comment and changing an existing comment fail with this error:

```
Error in solr.res$response$docs[[1]] : subscript out of bounds
```

The error appears in the session pane. 

Please check.

**Comment by prateek05** [comment]
Is it happening after 4bd1c9eebfe7e91bb3951c00bd4dbf0348be7400. 
I am not able to reproduce it with that patch included

**Comment by gordonwoodhull** [comment]
Right, the other bug (#2225) was masking this bug in the case of adding a new comment. 

Changing an existing comment would be unchanged. You can see the failure on rcloud-5.

Could it be a solr versioning problem?

**Comment by prateek05** [comment]
Okay I see what you mean now. No its not with the version. 
The bug occurs say if you manipulated comments on an instance and then load the gist in another (with a different solr) and manipulate it again.

If you work one a new gist from scratch are you still able to reproduce it?

**Comment by prateek05** [comment]
Its intermittent 
`solr.res$response$docs[[1]]$comments[[index]]` Need to understand the trigger. The error occurs from the response object

**Comment by prateek05** [comment]
Traced it to here: I am not familiar with this line of code. 
https://github.com/att/rcloud/blob/release/1.6.1/rcloud.support/R/search.R#L216
The error occurs when SOLR does not have the 'comment id' in its index. the grep returns null and the line following it crashes. 

What should be the expected behavior if the comment id is not found in the solr index ?
 - Drop the current index and rebuild a new one from the current state ?
 - Only add the comment id not found section to the index ? (Other entries will still remain out of sync from the current state of the gist) ?

**Comment by gordonwoodhull** [comment]
You're right, it doesn't happen for a new notebook. 

Maybe this occurred because we don't index the comments when we reindex all the notebooks? At least, it doesn't look like `update.solr` does anything comment-related. But I also see some old notebooks fail/not fail, so hmm.

I am not sure of the structure of the solr database, so I'm not sure what are the ramifications of these choices.

**Comment by prateek05** [comment]
yes we don't index comments on update.solr. The notebook object doesn't have the contents of the comments.

1. If error occurs while modifying we could fall back with adding the comment

A. Should update.solr , update comments too ? That would be slightly a big change anything calling update.solr would have to pass the contents of the comment too OR I can pull from github , its just that github calls are costly with response time. 

(Another issue i see is with multiline comments, search results come in as with div blocks, will raise another issue for that)

**Comment by gordonwoodhull** [comment]
No, `update.solr` shouldn't update comments, but our reindexing script probably should. 

I think that's unrelated to this issue; I was just guessing why add and modify fail on some notebooks but not others.

Falling back to adding the comments seems fine to me. But do note that add is also failing for the same notebooks. I tried using `numResults` instead of `docs[[1]]` for add, and that seemed to work, but then noticed that modify was also failing.

**Comment by prateek05** [comment]
I am rewriting that module. Calculation of index should not be based on a grep across the content of the comment.

**Comment by prateek05** [comment]
Done. Up for review

**Comment by gordonwoodhull** [comment]
Fixed in #2235
